src/wallet/wallet_rpc_server.cpp

```diff
@@ -635,6 +642,18 @@ namespace tools
         tx_to_blob(ptx_vector.back().tx, blob);
         res.tx_blob = epee::string_tools::buff_to_hex_nodelimer(blob);
       }
+      if (req.get_tx_metadata)
+      {
+        std::ostringstream oss;
+        binary_archive<true> ar(oss);
+        ::serialization::serialize(ar, const_cast<tools::wallet2::pending_tx&>(ptx_vector.back()));
+        cryptonote::blobdata metadata = epee::string_tools::buff_to_hex_nodelimer(oss.str());
+        res.tx_metadata_appended = !res.tx_blob.empty() && starts_with(metadata, res.tx_blob);
```

Comment:
This logic doesn't make any sense. If the user sets `get_tx_hex` and `get_tx_metadata`, this will serialize the transaction twice and the extra "metadata" once. It will then drop all of the additional metadata. Why not have the flags work independently? The caller can save serialization/deserialization time by only setting one of the flags. Or the backend can be more aggressive and only one to be set at a time.

Reply:
This is to not transmit the same data twice. If both are requested, the metadata only includes the suffix. Having both can be useful I think.

Reply:
D'oh, I thought the `std::string::substr` behavior was the opposite (return `0` to `pos`). A small optimization, `std::string::erase` will do a `memmove` instead of a new buffer allocation + `memcpy` that substr is doing.

I think that either the metadata blob should _never_ contain the `tx` field (which sort of ruins the serialization of `pending_tx` OR that the field should always be serialized with the metadata blob. The caller can decide if they want to lower their bandwidth by toggling the two available flags. I think the long-term maintenance effort will be easier than the currently "unique" implementation.